app/cli/internal/action/attestation_init.go

```diff
@@ -242,6 +220,96 @@ func (action *AttestationInit) Run(ctx context.Context, opts *AttestationInitRun
 	return attestationID, nil
 }
 
+// createWorkflow creates a new workflow if it doesn't exist, as well as its contract
+func (action *AttestationInit) createWorkflow(ctx context.Context, opts *AttestationInitRunOpts) (*WorkflowItem, error) {
+	// 1. find workflow if exists
+	wf, err := NewWorkflowDescribe(action.ActionsOpts).Run(ctx, opts.WorkflowName, opts.ProjectName)
+	if err != nil && status.Code(err) != codes.NotFound {
+		return nil, fmt.Errorf("error looking for workflow %q: %w", opts.WorkflowName, err)
+	}
+
+	contractRef := opts.NewWorkflowContractRef
+	contractName := contractRef
+	if contractRef != "" {
+		if isContractReference(contractRef) {
+			_, err := NewWorkflowContractDescribe(action.ActionsOpts).Run(contractRef, 0)
+			if err != nil {
+				return nil, fmt.Errorf("contract %q could not be found", contractRef)
+			}
+			// the contract exists
+		} else {
+			// use a default name for the contract
+			contractName = defaultContractName(opts.ProjectName, opts.WorkflowName)
+
+			// if the workflow exists, we just update its contract with the new contents
+			if wf != nil {
+				_, err = NewWorkflowContractUpdate(action.ActionsOpts).Run(wf.ContractName, nil, contractRef)
+				if err != nil {
+					return nil, fmt.Errorf("error updating contract %q: %w", wf.ContractName, err)
+				}
+				contractName = wf.ContractName
+			} else {
+				// if the workflow doesn't exist, let's create or update the contract
+				cont, err := NewWorkflowContractDescribe(action.ActionsOpts).Run(contractName, 0)
+				switch {
+				case err != nil && status.Code(err) == codes.NotFound:
+					// Contract not found, let's create it
+					_, err = NewWorkflowContractCreate(action.ActionsOpts).Run(contractName, nil, contractRef)
+					if err != nil {
+						return nil, err
+					}
+				case err != nil:
+					return nil, err
+				default:
+					// contract found, let's update it (chainloop will validate that there is an actual change in the contract file)
+					_, err := NewWorkflowContractUpdate(action.ActionsOpts).Run(cont.Contract.Name, &cont.Contract.Description, contractRef)
+					if err != nil {
+						return nil, err
+					}
+				}
+			}
+		}
+	}
+
+	// if workflow doesn't exist, let's create it with the contract
+	if wf == nil {
+		wf, err = NewWorkflowCreate(action.ActionsOpts).Run(&NewWorkflowCreateOpts{
+			Name:         opts.WorkflowName,
+			Project:      opts.ProjectName,
+			ContractName: contractName,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("error creating workflow %q: %w", opts.WorkflowName, err)
+		}
+	}
+
+	return wf, nil
+}
+
+// isContractReference checks if the reference points to an existent contract name (DNS-1123
+func isContractReference(ref string) bool {
```

Comment:
can't this method just check if is `valid DNS-1123`? 

also, we should have that validation error in biz